raytrace.py

```diff
@@ -101,7 +101,7 @@ def TW_absorptance_ratio(normal, b_constant, c_constant, incident):
 	
 
 	# noinspection PyUnresolvedReferences,PyUnresolvedReferences
-def refraction(incident, normal, n1, n2):
+def refraction(incident, normal, n1, n2, perpendicular_polarized):
```

Comment:
I think you should make the last parameter optional, taking False as its default value